ledger-core/v2/virtual/object/object.go

```diff
@@ -194,3 +199,16 @@ func (sm *SMObject) stepReadyToWork(ctx smachine.ExecutionContext) smachine.Stat
 func (sm *SMObject) stepWaitIndefinitely(ctx smachine.ExecutionContext) smachine.StateUpdate {
 	return ctx.Sleep().ThenRepeat()
 }
+
+func (sm *SMObject) createWaitPendingOrderedSM(ctx smachine.ExecutionContext) {
+	syncSM := AwaitOrderedPendingSM{
+		sync: sm.MutableExecute,
+	}
+
+	// syncSM acquire MutableExecute semaphore in init step.
+	ctx.InitChild(func(ctx smachine.ConstructionContext) smachine.StateMachine {
+		return &syncSM
+	})
+
+	sm.AwaitPendingOrdered = syncSM.stop
```

Comment:
@cyraxred, is it allowed to copy barge in like this from other state machine?

Reply:
@ruz As I can see in player.go BargIn copied to shared data like this
<img width="390" alt="Screenshot 2020-05-14 at 11 54 03" src="https://user-images.githubusercontent.com/21306401/81914213-af513c80-95d9-11ea-8a9b-6c613743d268.png">
I can change to a pointer if you not like this copy)

Reply:
it's different. In the code actually to cases: [case 1](https://github.com/insolar/assured-ledger/blob/fdad6296dafbb8f926036555a0d790023bac2fa1/ledger-core/v2/conveyor/smachine/example/player.go#L112) and [case 2](https://github.com/insolar/assured-ledger/blob/fdad6296dafbb8f926036555a0d790023bac2fa1/ledger-core/v2/conveyor/smachine/example/player.go#L135). In both cases bargeins are injected into shared data. In case one it's access to shared data. In case two it's publish of a new shared data.

Reply:
Got it.

Reply:
@ruz Fixed

Reply:
@ruz actually, bargein, like synclink is by-value, so no need for ptr